src/cryptonote_core/service_node_list.cpp

```diff
@@ -723,11 +718,16 @@ namespace service_nodes
     // and if we don't already have the maximum
     auto contrib_iter = std::find_if(contributors.begin(), contributors.end(),
         [&address](const service_node_info::contribution& contributor) { return contributor.address == address; });
-    if (contrib_iter == contributors.end())
-    {
-      if (contributors.size() >= MAX_NUMBER_OF_CONTRIBUTORS || transferred < info.get_min_contribution())
-        return;
-    }
+
+    const bool contributer_found = (contrib_iter != contributors.end());
+
+    /// Only create a new contributor if we don't exceed the contributor limit
+    if (!contributer_found && contributors.size() >= MAX_NUMBER_OF_CONTRIBUTORS) return;
+
+    /// Check that the contribution is large enough
+    const uint8_t hf_version = m_blockchain.get_hard_fork_version(block_height);
+    const uint64_t min_contribution = get_min_node_contribution(hf_version, info.staking_requirement, info.total_reserved, contributors.size());
+    if (!contributer_found && transferred < min_contribution) return;
 
     m_rollback_events.push_back(std::unique_ptr<rollback_event>(new rollback_change(block_height, pubkey, info)));
 
```

Comment:
The line below, `contrib_iter == contributors.end()` can be replaced with the `new_contributor` or `!contributor_found`.